quickwit-cli/src/lib.rs

```diff
@@ -393,6 +407,39 @@ fn display_statistics(
     Ok(())
 }
 
+/// ThroughputCalculator is used to calculate throughput.
+struct ThroughputCalculator {
+    /// Stores the time series of processed bytes value.
+    processed_bytes_values: VecDeque<(Instant, usize)>,
+}
+
+impl ThroughputCalculator {
+    /// Creates new instance.
+    pub fn new(now: Instant) -> Self {
+        let processed_bytes_values: VecDeque<(Instant, usize)> =
+            (0..THROUGHPUT_WINDOW_SIZE).map(|_| (now, 0usize)).collect();
+        Self {
+            processed_bytes_values,
+        }
+    }
+
+    /// Calculates the throughput.
+    pub fn calculate(&mut self, current_processed_bytes: usize) -> f64 {
+        self.processed_bytes_values.pop_front();
+        let current_instant = Instant::now();
+        let (first_instant, first_processed_bytes) = *self.processed_bytes_values.front().unwrap();
+        let elapsed_time = (current_instant - first_instant).as_secs();
+        self.processed_bytes_values
+            .push_back((current_instant, current_processed_bytes));
+
+        //TODO: what should we do when all docs are processed and `current_processed_bytes`
```

Comment:
@guilload  I need your opinion here. 
So the `processed_bytes` is the number or byte ingested from documents when indexing. From the point the last document is indexed to the time the last split is uploaded, the `processed_bytes` no longer increases making the throughput drop gradually to zero. 
is this drop ok to show to the user or could we just display that last valid throughput value for the rest of the time?

Reply:
This is a good thing. That's what the  user needs to see.

Reply:
Yes, that's fine if we display that last valid throughput value after the last document has been ingested. However, to Paul's point, we need to display something else to inform the user that the program is actually doing something and not hanging.

Reply:
Trying to come up with another metric, I have added the elapsed_time for now since that one changes. 
```bash
quickwit index --index-uri 'file://home/evance/quickwit/data' --input-path ./zero/wiki-articles-10000.json
Num docs:    9970 Parse errs:     0 Staged splits:   0 Input size:    11MB Thrghput: 11.18MB/s Time: 00:00:01
Num docs:   10000 Parse errs:     0 Staged splits:   0 Input size:    11MB Thrghput:  5.62MB/s Time: 00:00:02
Num docs:   10000 Parse errs:     0 Staged splits:   0 Input size:    11MB Thrghput:  3.74MB/s Time: 00:00:03
Num docs:   10000 Parse errs:     0 Staged splits:   0 Input size:    11MB Thrghput:  2.81MB/s Time: 00:00:04
Num docs:   10000 Parse errs:     0 Staged splits:   0 Input size:    11MB Thrghput:  0.02MB/s Time: 00:00:05
Num docs:   10000 Parse errs:     0 Staged splits:   1 Input size:    11MB Thrghput:  0.00MB/s Time: 00:00:05

Indexed 10000 documents in 5s
You can now query your index with `quickwit search --index-uri file://home/evance/quickwit/data --query "barack obama"`
```